### PR TITLE
Adding backwards compatibility for name identity

### PR DESCRIPTION
**Pull request by galvana**
### Description Of Changes

Part of the work I did when adding custom identities was to demote "name" as an identity since it's not unique enough to uniquely identify a subject. This PR adds support to render the name field (if provided in the Privacy Center's `config.json`) but ignore it as part of the privacy request payload.

I'm also fixing a test with the demo seed data, I'm including it here to minimize the time spent juggling two separate PRs.

### Code Changes

* [ ] Updated `PrivacyRequestForm.tsx` component
* [ ] Fixed `test_seed.py` tests

### Steps to Confirm

* [ ] Run `nox -s dev` to start the Fides server
* [ ] Start the Privacy Center `cd clients/privacy-center && turbo run dev`
* [ ] You should be able to specify `name` in the `identity_inputs` of the Privacy Center's `config.json` and still be able to submit a request

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 1]
```diff
@@ -64,7 +65,15 @@ const usePrivacyRequestForm = ({
...
       ...Object.fromEntries(
-        Object.entries(identityInputs).map(([key]) => [key, ""])
+        Object.entries(identityInputs)
+          .filter(
+            ([key, value]) =>
+              key === "name" ||
+              key === "phone" ||
+              key === "email" ||
+              (typeof value === "object" && value.label)
+          )
```
Comment (galvana) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:75]: Only keeping the known identities (name, phone, email) or custom identities with a label, this is to ignore something just doing
```
"loyality_id: "required"
```
it needs to be labeled
```
"loyalty_id": {"label" "Loyalty ID"}
```

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 2]
```diff
@@ -80,13 +89,21 @@ const usePrivacyRequestForm = ({
...
+          // we have to support name as an identity_input for legacy purposes
+          // but we ignore it since it's not unique enough to be treated as an identity
+          .filter(([key]) => key !== "name")
+          .map(([key, field]) => {
+            const value = fallbackNull(values[key]);
+            if (typeof field === "string") {
+              if (key === "phone") {
+                // eslint-disable-next-line no-param-reassign
+                key = "phone_number";
+              }
```
Comment (galvana) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:102]: The key is `phone` in `config.json` but it needs to be converted to `phone_number` to be posted

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 3]
```diff
@@ -80,13 +89,21 @@ const usePrivacyRequestForm = ({
...
-          const value = fallbackNull(values[key]);
-          if (typeof field === "string") {
-            return [key, value];
-          }
-          return [key, { label: field.label, value }];
-        })
+        Object.entries(action.identity_inputs ?? {})
+          // we have to support name as an identity_input for legacy purposes
+          // but we ignore it since it's not unique enough to be treated as an identity
+          .filter(([key]) => key !== "name")
```
Comment (galvana) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:95]: The `name` field will appear in the form but be ignored from the privacy request payload

File: tests/ctl/api/test_seed.py
Comment (galvana) [inline tests/ctl/api/test_seed.py:1]: Fixing seed test

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 4]
```diff
@@ -211,7 +229,13 @@ const usePrivacyRequestForm = ({
...
       ...Object.fromEntries(
         Object.entries(identityInputs)
-          .filter(([key]) => key !== "email" && key !== "phone")
+          .filter(
+            ([key, value]) =>
+              key !== "email" &&
+              key !== "phone" &&
+              key !== "name" &&
+              typeof value !== "string"
+          )
```
Comment (galvana) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:238]: Filter out the default identities or identity inputs with just string values (such as "required" or "optional") from the dynamic form validation

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 5]
```diff
@@ -306,29 +330,80 @@ const PrivacyRequestForm: React.FC<PrivacyRequestFormProps> = ({
...
+              </FormControl>
+            ) : null}
+            {Object.entries(identityInputs)
+              .filter(
+                ([key, item]) =>
+                  key !== "email" &&
+                  key !== "phone" &&
+                  key !== "name" &&
+                  typeof item !== "string"
+              )
```
Comment (galvana) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:398]: Same filtering to render the form fields

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx (hunk 1 above)
Comment (NevilleS) [reply clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:75]: The repeated list of keys here would benefit from some kind of constant declaration like:
```
const DEFAULT_IDENTITY_INPUTS_MAP = {
  "name": undefined,
  "email": "email",
  "phone": "phone_number",
};
```
...which could more easily do comparisons like `Object.keys(DEFAULT_IDENTITY_INPUTS_MAP).includes("name")` etc

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx (hunk 3 above)
Comment (NevilleS) [reply clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:95]: 👍 

OK, this'll work for our purposes. It's pretty confusing behaviour to explain to someone new to the code, but it's also confusing as is, so this is an improvement!

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx (hunk 2 above)
Comment (NevilleS) [reply clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:102]: Oof. That's messy. The `IDENTITY_INPUTS_MAP` suggestion I had above was to try to help make this _slightly_ less magical, but it's still a confusing DX

**Review by adamsachs — APPROVED** [review]
code looks enough to my novice eyes, and manual testing with `nox -s "fides_env(test)"` is looking good!

image

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx (hunk 5 above)
Comment (NevilleS) [reply clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:398]: nit: You'll find this kind of rendering logic easier to manage if you pre-filter this above, like
```
const defaultIdentityInputs = Object.entries(identityInputs).filter(...)
const customIdentityInputs = Object.entries(identityInputs).filter(...)
```

And then you can use those variables directly with less fuss 😄

**Review by NevilleS — APPROVED** [review]
Couple nits but not particularly worth spending energy on here 😄

**Pull request merged**